extensions/transaction/transaction-atomikos/src/main/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionContext.java

```diff
@@ -66,17 +66,49 @@ public void execute(TransactionBlock block) {
             }
             throw new EdcException(e);
         } finally {
-            if (startedTransaction) {
-                try {
-                    var status = transactionManager.getStatus();
-                    if (STATUS_ACTIVE == status) {
-                        transactionManager.commit();
-                    } else if (STATUS_MARKED_ROLLBACK == status) {
-                        transactionManager.rollback();
-                    }
-                } catch (HeuristicRollbackException | SystemException | HeuristicMixedException | RollbackException e) {
-                    monitor.severe("Transaction error", e);
+            commitOrRollback(startedTransaction);
+        }
+    }
+
+    @Override
+    public <T> T execute(ResultTransactionBlock<T> block) {
```

Comment:
Same as the comment on the `LocalTransactionContext`, and probably the no result execute implementation could be set as `default` on the interface.